pkg/filebacked/file.go

```diff
@@ -72,23 +83,43 @@ func (w *Writer) Reader() (reader *Reader) {
 	err := w.file.Sync()
 	if err == nil {
 		err = os.Link(w.path, path)
+		if err != nil {
+			log.Error(err, "link failed.")
+		}
 	}
 	reader = &Reader{
 		error: liberr.Wrap(err),
 		path:  path,
 	}
+	runtime.SetFinalizer(
+		reader,
+		func(r *Reader) {
+			r.Close()
+		})
```

Comment:
Add finalizer to ensure the hard-link is deleted in cases where the reader (iterator) is never used.